lib/vagrant-openshift/hooks/find_ami.rb

```diff
@@ -0,0 +1,42 @@
+#--
+# Copyright 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+require_relative "../aws"
+
+module Vagrant
+  module Openshift
+    module Hooks
+      class FindAMI
+        def initialize(app, env)
+          @app = app
+        end
+
+        def call(env)
+          if env[:machine].config.openshift.autoconfigure_aws == true
+            aws = env[:machine].config.vm.get_provider_config(:aws)
+            if !aws.nil? && !aws.ami.nil? && !aws.ami.start_with?("ami-")
+              box_info = YAML.load_file(Pathname.new(File.expand_path("#{__FILE__}/../../templates/command/init-openshift/box_info.yaml")))
```

Comment:
No, I'm specifically not touching `.vagrant-openshift.json` because it doesn't really help my use case and on the understanding that it was probably going away anyway.